src/repository/BaseRepository.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+/**
+ * Base class for repository pattern
+ */
+let self;
+
+export default class BaseRepository {
+    constructor(q, config, constants) {
+        self = this;
+        self.q = q;
+        self.config = config;
+        self.constants = constants;
+    }
+
+    _insert(object) {
```

Comment:
@AshanthaLahiru Can you get insert to take in just the object (instead of the model) from the child repository? Create the model object here.